modules/reactor/include/meshgenerators/PolygonMeshGeneratorBase.h

```diff
@@ -147,17 +150,140 @@ class PolygonMeshGeneratorBase : public MeshGenerator
+                    dof_id_type & node_id_background_meta,
+                    const Real azimuthal_angle,
+                    const std::vector<Real> azimuthal_tangent,
+                    const unsigned int side_index,
+                    const bool quad_center_elements,
+                    const Real rotation_angle);
+
+  /**
+   * Generates a mesh of a slice, which is the foundation of both buildGeneralSlice and
+   * buildSimpleSlice.
+   * @param ring_radii radii of the ring regions
+   * @param ring_layers numbers of radial intervals of the ring regions
+   * @param ring_radial_biases values used for radial meshing biasing in ring regions
+   * @param ring_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the ring regions
+   * @param ring_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the ring regions
+   * @param ducts_center_dist distance parameters of the duct regions
+   * @param ducts_layers numbers of radial intervals of the duct regions
+   * @param duct_radial_biases values used for radial meshing biasing in duct regions
+   * @param duct_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the duct regions
+   * @param duct_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the duct regions
+   * @param pitch twice of the length of the first side times cosine of the azimuthal angle
+   * @param num_sectors_per_side number of azimuthal intervals
+   * @param background_intervals number of radial intervals of the background region
+   * @param background_radial_bias value used for radial meshing biasing in background region
+   * @param background_inner_boundary_layer_params width, radial sectors, and growth factor of the
+   * inner boundary layer of the background region
+   * @param background_outer_boundary_layer_params width, radial sectors, and growth factor of the
+   * outer boundary layer of the background region
+   * @param node_id_background_meta pointer to the first node's id of the background region
+   * @param virtual_side_number 360.0 over the azimuthal angle of the slice (happens to be number of
+   * sides of the polygon if a polygon is to be generated)
+   * @param side_index index of the polygon side
+   * @param azimuthal_tangent vector of tangent values of the azimuthal angles as reference for
+   * adaptive boundary matching
+   * @param block_id_shift shift of the subdomain ids generated by this function
+   * @param quad_center_elements whether the central region contrains quad elements or not
+   * @param boundary_id_shift shift of the interface boundary ids
+   * @param pitch_scale_factor the ratio between the second side to the first side.
```

Comment:
which sides?
This is when the polygon has a degree of asymmetry?

Reply:
This is the primary and secondary sides defined in `buildGeneralSlice()`. Will revise to make sure they are consistent.